libpdb/request_manager.go

```diff
@@ -111,43 +132,45 @@ func (rm *RequestManager) writePeriodic() {
 func (rm *RequestManager) readPeriodic() {
 
 	for rm.isDead() == false {
-		select {
-		case msg := <-rm.readChan:
-			rm.readQueue = append(rm.readQueue, msg)
-		default:
-			config := rm.config.Load().(ClientConfig)
-			var req *common.ReadRequest = nil
-			var args *common.ReadArgs
-			var reply common.ReadReply
-			if len(rm.readQueue) > 0 {
-				req = rm.readQueue[0]
-				args = req.Args
-				rm.readQueue = rm.readQueue[1:]
-				rm.log.Info.Printf("readPeriodic: Real request \n")
-			} else {
-				args = &common.ReadArgs{}
-				rm.generateRandomRead(config, args)
-				rm.log.Info.Printf("readPeriodic: Dummy request \n")
-			}
-			//@todo Do something with response
-			startTime := time.Now()
-			err := rm.leader.Read(args, &reply)
-			elapsedTime := time.Since(startTime)
+		var args *common.ReadArgs
+		var replier RequestResponder
+		reply := common.ReadReply{}
 
-			if err != nil {
-				reply.Err = err.Error()
-			}
-			if reply.Err != "" {
-				rm.log.Error.Printf("readPeriodic error: %v, reply=%v, time=%v\n", err, reply, elapsedTime)
-			} else {
-				rm.log.Info.Printf("readPeriodic reply: range=%v, time=%v\n", reply.GlobalSeqNo, elapsedTime)
-			}
+		if rm.RequestGenerator != nil {
+			args, replier = rm.RequestGenerator.NextRequest()
+		}
 
-			if req != nil {
-				req.Reply(&reply)
-			}
-			time.Sleep(config.ReadInterval)
+		config := rm.config.Load().(ClientConfig)
+
+		if args != nil {
+			rm.log.Info.Printf("readPeriodic: Real request \n")
+		} else {
+			args = &common.ReadArgs{}
```

Comment:
In retrospect, there's no value in generating random reads right? We mind as well always be polling for something we care about